core/src/main/java/uk/ac/bbsrc/tgac/miso/core/security/util/LimsSecurityUtils.java

```diff
@@ -72,24 +81,38 @@ public static User fromLdapUser(InetOrgPerson details) {
     // user and this user will inherit the already-persisted userID.
     final UserImpl user = new UserImpl();
 
-    user.setActive(details.isAccountNonExpired());
-    user.setAdmin(details.getAuthorities().contains(new GrantedAuthorityImpl("ROLE_ADMIN")));
-    user.setExternal(details.getAuthorities().contains(new GrantedAuthorityImpl("ROLE_EXTERNAL")));
-    user.setInternal(details.getAuthorities().contains(new GrantedAuthorityImpl("ROLE_INTERNAL")));
-    user.setLoginName(details.getUsername().toLowerCase());
-    user.setPassword(details.getPassword());
-    user.setFullName(details.getDisplayName());
-    user.setEmail(details.getMail());
-
     final List<String> roles = new ArrayList<String>();
     for (final GrantedAuthority ga : details.getAuthorities()) {
-      roles.add(ga.toString());
+      roles.add(removePrefix(ga.toString(), LimsSecurityUtils.rolePrefix));
     }
     user.setRoles(roles.toArray(new String[0]));
 
+    user.setActive(details.isAccountNonExpired());
+    user.setAdmin(roles.contains(new GrantedAuthorityImpl("ROLE_ADMIN")));
+    user.setExternal(roles.contains(new GrantedAuthorityImpl("ROLE_EXTERNAL")));
+    user.setInternal(roles.contains(new GrantedAuthorityImpl("ROLE_INTERNAL")));
+    user.setLoginName(details.getUsername().toLowerCase());
+    if (details.getPassword() == null) {
+      log.debug("This directory has been configured correctly and has properly returned a null password. Setting fake password "
```

Comment:
So, the database is set up to allow null passwords, but the SHA encoder doesn't do a null check. I'm wondering if it's better to fix that code (or use the passthrough encoder, which does no checks). Or did you see other errors?